CONTRIBUTING.md

```diff
@@ -0,0 +1,78 @@
+# Contributing
+
+Welcome and thank you for wanting to contribute to the Trustlines Project.
+We appreciate all help, depending on your knowledge, this can be
+spreading the word, helping with documentation, reporting issues or even contributing to the source code.
+
+Feel free to checkout the code, play around with it and run the tests.
+Before making a change, please first discuss the change via opening an issue on this
+repository. If you just need some small question answered, you can also reach out to
+us via our [gitter chat](https://gitter.im/trustlines/community).
+
+## Open an issue
+
+To request a feature or report a bug, please open an issue on the issue tracker in github.
+
+### Report a bug
+
+Please include a short description of the problem, including what you expected to happen and what did happen instead.
+Please also include the version of the clientlib you were using, your operating system and the used configuration.
+
+### Request a feature
+
+Please include a description of what you would like to see implemented and an explanation of why you believe this would
+be a good addition.
+
+## Open a pull request
+
+### Before starting
+
+- Follow the [readme](/README.md) to set you dev environment up.
+- Ensure that you can run the tests and that they pass.
+- Ensure that you can run the end2end tests. You can find more information in the
+  [end2end repo](https://github.com/trustlines-protocol/end2end)
+
+### To include
+
+- For a bug fix:
+  - Add a test that makes the bug explicit, and make sure that the test fails.
+  - Fix the bug, so that the test passes.
+- For a new feature:
+  - Add a test, that demonstrate the new feature.
+  - Add or update related documentation if necessary.
+- Add an entry to the unreleased section of the [changelog](CHANGELOG.md).
+- Format your code changes with prettier. If you setup pre-commit, this will happen automatically on commit.
+
+## Changelog
+
+We loosely follow [keep a changelog](https://keepachangelog.com/en/0.3.0/).
```

Comment:
On this repo we follow this actually more strict, so you might want to change the wording here from `add` to `added` for example, and they are only grouped, so do not have to start with the keyword.